R/init.R

```diff
@@ -222,27 +186,246 @@ init <- function(data,
+    lifecycle::deprecate_warn(
+      when = "0.15.3",
+      what = "init(header)",
+      details = paste(
+        "Use `modify_header()` on the teal app object instead."
       )
-    },
-    server = function(input, output, session) {
-      if (!is.null(landing_popup)) {
-        do.call(landing_popup$server, c(list(id = "landing_module_shiny_id"), landing_popup$server_args))
-      }
-      srv_teal(id = ns("teal"), data = data, modules = modules, filter = deep_copy_filter(filter))
+    )
+  }
+  if (!is.null(footer)) {
+    checkmate::assert_multi_class(footer, c("shiny.tag", "shiny.tag.list", "html", "character"))
+    lifecycle::deprecate_warn(
+      when = "0.15.3",
+      what = "init(footer)",
+      details = paste(
+        "Use `modify_footer()` on the teal app object instead."
+      )
+    )
+  }
+
+  # argument transformations
+  ## `modules` - landing module
+  landing <- extract_module(modules, "teal_module_landing")
+  if (length(landing) == 1L) {
+    res$landing_popup_server <- landing[[1L]]$server
+    modules <- drop_module(modules, "teal_module_landing")
+    lifecycle::deprecate_warn(
+      when = "0.15.3",
+      what = "landing_popup_module()",
+      details = paste(
+        "landing_popup_module() is deprecated.",
+        "Use add_landing_popup() on the teal app object instead."
+      )
+    )
+  } else if (length(landing) > 1L) {
+    stop("Only one `landing_popup_module` can be used.")
+  }
+
+  # Note: UI must be a function to support bookmarking.
+  res$ui <- function(request, ...) {
+    ui_teal(
+      id = "teal",
+      modules = modules,
+      title = title,
+      header = header,
+      footer = footer
+    )
+  }
+  res$server <- function(input, output, session) {
+    if (!is.null(res$landing_popup_server)) {
+      do.call(res$landing_popup_server, c(list(id = "landing_module_shiny_id")))
     }
-  )
+    srv_teal(id = "teal", data = data, modules = modules, filter = deep_copy_filter(filter))
+  }
 
   logger::log_debug("init teal app has been initialized.")
 
   res
 }
+
+#' Add a custom title to the app
+#'
+#' @param title The title to add
+#' @return The modified app object
+#' @export
+#' @examples
+#' app <- init(
+#'   data = teal_data(IRIS = iris, MTCARS = mtcars),
+#'   modules = modules(example_module())
+#' ) |>
+#'   modify_title("Custom title")
+#'
+#' shinyApp(app$ui, app$server)
+modify_title <- function(
+    app,
+    title = "teal app",
+    favicon = "https://raw.githubusercontent.com/insightsengineering/hex-stickers/main/PNG/nest.png") {
+  res <- app
+  res$ui <- function(request, ...) {
+    args <- list(...)
+    args$title <- tags$div(
+      id = "teal-title",
+      tags$head(
+        tags$title(title),
+        tags$link(
+          rel = "icon",
+          href = favicon,
+          sizes = "any"
+        )
+      )
+    )
+    ui_tq <- do.call(app$ui, c(list(request = request), args)) |>
+      htmltools::tagQuery()
```

Comment:
No pipe before changing R version in Description file
```suggestion
    ui_tq <- htmltools::tagQuery(do.call(app$ui, c(list(request = request), args)))
```